presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -769,6 +736,41 @@ public void revokeTablePrivileges(String databaseName, String tableName, String
         }
     }
 
+    private PrivilegeBag buildPrivilegeBag(HiveMetastoreClient metastoreClient,
+                                                     String databaseName,
+                                                     String tableName,
+                                                     String grantee,
+                                                     Set<PrivilegeGrantInfo> privilegeGrantInfos)
+    {
+        PrincipalType principalType;
+        String principalName;
+
+        try {
+            if (metastoreClient.getRoleNames().contains(grantee.toLowerCase())) {
```

Comment:
If the grantee is `PUBLIC`, can we skip the call to load the roles since it is a reserved name?